braindecode/datasets/base.py

```diff
@@ -99,40 +99,41 @@ class BaseConcatDataset(ConcatDataset):
     """A base class for concatenated datasets. Holds either mne.Raw or
     mne.Epoch in self.datasets and has a pandas DataFrame with additional
     description.
-
     Parameters
     ----------
     list_of_ds: list
-        list of BaseDataset of WindowsDataset to be concatenated.
+        list of BaseDataset, BaseConcatDataset or WindowsDataset
     """
     def __init__(self, list_of_ds):
+        # if we get a list of BaseConcatDataset, get all the individual datasets
+        if isinstance(list_of_ds[0], BaseConcatDataset):
+            list_of_ds = [d for ds in list_of_ds for d in ds.datasets]
         super().__init__(list_of_ds)
         self.description = pd.DataFrame([ds.description for ds in list_of_ds])
+        self.description.reset_index(inplace=True, drop=True)
```

Comment:
Is there a reason to not reset the index of the new pandas DataFramge and keep the original index somewhere?